app/javascript/controllers/pair_requests_form_controller.js

```diff
@@ -0,0 +1,62 @@
+import { Controller } from "@hotwired/stimulus"
+export default class extends Controller {
+  static targets = ["click", "invitee", "invitee_tz"]
+  connect(){
+    localStorage.clear()
+    console.log("localStorage is cleared")
+  }
+
+  click(event) {
+    const invitee_time_zone = event.target.selectedOptions[0].dataset.timezone
+    localStorage.setItem("invitee_time_zone", invitee_time_zone)
+
+    if(localStorage.getItem("pair_app_when")){
+      let time = localStorage.getItem("pair_app_when")
+      time = new Date(time);
+
+      this.setSchedule(invitee_time_zone, time)
+    }
+  }
+  
+  date(event){
+    let time = new Date(event.target.value)
+    localStorage.setItem("pair_app_when", time)
+    if(localStorage.getItem("invitee_time_zone")){
+      const inviteeValue = localStorage.getItem("invitee_time_zone")
+      this.setSchedule(inviteeValue, time)
+    }
+  }
+
+  setSchedule(invitee, time){
+    const inviteeResult = new TimeZoneConverter(invitee, time).conversion
+    this.invitee_tzTarget.textContent = (`(${invitee})`)
+    this.inviteeTarget.textContent = (`${inviteeResult}`)
+  }
+}
+
+class TimeZoneConverter {
+  constructor(input, time) {
+    this.input = input;
+    this.time = time
+  }
+
+  get conversion() {
+    let match
+    let ary = Intl.supportedValuesOf('timeZone');
+    if (this.input == 'UTC'){
+      match = this.input
```

Comment:
is this check needed?? Doesn't the else just handle this?

Reply:
'UTC' seemed to be a weird edge case because it's not in JS's `ary = Intl.supportedValuesOf('timeZone');` and without the check the conversion wouldn't be properly set.